### PR TITLE
Add LangChain integration and enhance PromptVault functionality

### PR DESCRIPTION
**Pull request by mikehostetler**
Addresses #1 

- Implemented the `Enumerable` protocol for `PromptVault.Context`, allowing direct usage with LangChain functions.
- Added helper functions for converting between PromptVault and LangChain message formats.
- Enhanced documentation, including a new `docs/langchain_integration.md` guide with examples.
- Increased test coverage from 80.3% to 90.1%, adding comprehensive tests for all message types and edge cases.
- Refactored `PromptVault.LangChain` module to focus on integration features and improved error handling.

These changes significantly improve the interoperability of PromptVault with LangChain, providing a seamless experience for users.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,10 +13,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Comprehensive documentation and examples in `docs/langchain_integration.md`
 
 ### Changed
 - Updated mix.exs to include tiktoken dependency (~> 0.4.1)
 - Enhanced README.md with token counting section and installation instructions
+- Refactored `PromptVault.LangChain` module to focus on helper functions rather than wrapper API
+- Updated documentation to reflect direct integration approach
+
+### Improved
+- Test coverage increased from 80.3% to 90.1%
```
Comment (jacobluetzow) [inline CHANGELOG.md:31]: 🔥

**Review by jacobluetzow — APPROVED** [review]
